app/build.gradle

```diff
@@ -173,10 +172,10 @@ dependencies {
     implementation 'io.reactivex.rxjava2:rxandroid:2.1.1'
     implementation 'io.reactivex.rxjava2:rxjava:2.2.17'
 
-    implementation 'com.google.firebase:firebase-analytics:17.2.2'
-    implementation 'com.google.firebase:firebase-auth:19.2.0'
-    implementation 'com.google.firebase:firebase-functions:19.0.2'
-    implementation 'com.google.firebase:firebase-messaging:20.1.2'
+    implementation 'com.google.firebase:firebase-analytics:17.3.0'
+    implementation 'com.google.firebase:firebase-auth:19.3.0'
+    implementation 'com.google.firebase:firebase-functions-ktx:19.0.2'
+    implementation 'com.google.firebase:firebase-messaging:20.1.5'
```

Comment:
FCM SDK versions 20.1.4 and 20.1.5 have an issue that may lead to an increase in crashes due to shared resource exhaustion.

Reply:
should update to `20.1.6`

Reply:
You are right. Just read the [release note](https://firebase.google.com/support/release-notes/android#messaging_v20-1-5). But should it be `20.1.7`?